### DISCUSSION TITLE
how can i only display todos that are 'open' ?

### DISCUSSION THREAD

**Discussion by kristus123** [johnlindquist/kit#915] [Q&A]
i was looking at this tutorial, but it wasn't clear how the list was rendered to the gui, so i have no idea how i can filter it to only show 'open tasks'.

Maybe someone knows, thank you in advancd!

https://egghead.io/lessons/script-kit-build-a-multi-tab-todos-app-in-script-kit

**Comment by johnlindquist** [comment]
@kristus123 An array `.filter` will work just fine. Let me know if you'd like a more detailed explanation of any part of it:

```js
// Name: Todos with Open Tab

import "@johnlindquist/kit"

let { todos, write } = await db({ todos: [] })

onTab("Open", async () => {
  while (true) {
    let todo = await arg("Toggle todo:", todos.filter(t => t.tag === "open"))
    let t = _.find(todos, todo)
    t.tag = t.tag === "open" ? "done" : "open"
    await write()
  }
})

onTab("Done", async () => {
  while (true) {
    let todo = await arg("Toggle todo:", todos.filter(t => t.tag === "done"))
    let t = _.find(todos, todo)
    t.tag = t.tag === "open" ? "done" : "open"
    await write()
  }
})


onTab("New", async () => {
  let name = await arg("New todo:")
  todos.push({ name, id: uuid(), tag: "open" })
  await write()
  setTab("Open")
})

onTab("Remove", async () => {
  let todo = await arg("Remove todo:", todos)
  _.remove(todos, todo)
  await write()
  setTab("Open")
})
```

[Open example-todos in Script Kit](https://scriptkit.com/api/new?name=example-todos&url=https://gist.githubusercontent.com/johnlindquist/301a497ffdb50534f44e44fc4174ad9f/raw/ed9ec87360adc8457cf84de218f54a67cc984ee0/example-todos.ts")

**Reply by kristus123** [reply]
I appreciate your effort spent on this community. This is a tool that makes life better for many people. Not only do you make an awesome tool, but also spend time helping people

And thank you for answering my question ❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️

**Reply by johnlindquist** [reply]
Thanks for the kind words! Really makes my day 😊
